tests/TestCollectors.py

```diff
@@ -12,33 +12,52 @@
 from vrops_exporter import run_prometheus_server
 from tools.YamlRead import YamlRead
 from VropsCollector import VropsCollector
+from tools.Resources import Resources
+from resources.Vcenter import Vcenter
+
 
 class TestCollectors(unittest.TestCase):
     def test_environment(self):
-        self.assertTrue(os.getenv('USER'),'no dummy USER set')
-        self.assertTrue(os.getenv('PASSWORD'),'no dummy PASSWORD set')
+        self.assertTrue(os.getenv('USER'), 'no dummy USER set')
+        self.assertTrue(os.getenv('PASSWORD'), 'no dummy PASSWORD set')
 
     def test_collector_metrics(self):
         metrics_yaml = YamlRead('tests/metrics.yaml').run()
         print(metrics_yaml)
-        #every collector got to be tested in here
-        random_prometheus_port = random.randrange(9000,9700,1)
+
+        # every collector got to be tested in here
+        random_prometheus_port = random.randrange(9000, 9700, 1)
         print("chosen testport: " + str(random_prometheus_port))
         for collector in metrics_yaml.keys():
+            print()  # nicer output
             print("\nTesting " + collector)
+
+            Vcenter.add_datacenter = MagicMock()
+            VropsCollector.create_resource_objects = MagicMock(return_value=Vcenter)
+            VropsCollector.get_adapter = MagicMock(return_value='{"name": "vcenter1", "uuid": "5628-9ba1-55e84701"}')
+
             VropsCollector.get_modules = MagicMock(return_value=('/vrops-exporter/module', [collector]))
 
-            #start prometheus server to provide metrics later on
+            # test tool get_resources to create resource objects
+
+            Resources.get_datacenter = MagicMock(return_value={'name': 'datacenter1', 'uuid': '5628-9ba1-55e847050814'})
+            Resources.get_cluster = MagicMock(return_value={'name': 'cluster1', 'uuid': '3628-93a1-56e84634050814'})
+            Resources.get_hosts = MagicMock(return_value={'name': 'hostsystem1', 'uuid': '3628-93a1-56e84634050814'})
+            Resources.get_vmfolders = MagicMock(return_value={'name': 'vmfolder1', 'uuid': '3628-93a1-56e84634050814'})
+            Resources.get_virtualmachines = MagicMock(return_value={'name': 'vm1', 'uuid': '3628-93a1-56e8463404'})
+            Resources.get_resources = MagicMock(return_value={'name': 'vm1', 'uuid': '3628-93a1-56e8463404'})
+
```

Comment:
Cool! May you add more than one element into every mocked call? Would like to see the behavior tested with n+1 at least.

Reply:
resolved